ztoc/zinfo_builder.go

```diff
@@ -0,0 +1,115 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package ztoc
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/awslabs/soci-snapshotter/compression"
+	"github.com/opencontainers/go-digest"
+)
+
+// ZinfoBuilder builds the `zinfo` part of a ztoc. This interface should be
+// implemented for each compression algorithm we support.
+type ZinfoBuilder interface {
+	// ZinfoFromFile builds zinfo given a compressed tar filename and span size, and calculate the size of the file.
+	ZinfoFromFile(filename string, spanSize int64) (zinfo CompressionInfo, fs compression.Offset, err error)
```

Comment:
Can we please rename [CompressionInfo](https://github.com/awslabs/soci-snapshotter/blob/1e87baff6ad9fec046c39e3ed7c4245b40efea79/ztoc/ztoc.go#L59) to ``zInfo``? This naming inconsistency is confusing.

Reply:
Agree, especially we're calling them `zinfo` in most cases (e.g. docs). Will open an issue to track it separately since it requires more changes (cli, fs, ztoc and flatbuffer).